tools/prepare-curated.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Prepare the curated data and save the result to the given folder.
+ *
+ * Curation means copying raw data to the given folder, applying patches (CSS,
+ * elements, IDL) when needed, re-generating the `idlparsed` folder, and
+ * re-generating the `idlnames` and `idlnamesparsed` folders for relevant
+ * browser specs.
+ *
+ * The output folder gets created if it does not exist yet. Output folder
+ * contents get deleted to start with if folder is not empty.
+ *
+ * node tools/prepare-curated.js [raw data folder] [curated folder]
+ */
+
+const fs = require('fs').promises;
+const path = require('path');
+const rimraf = require('rimraf');
+const {
+  expandCrawlResult,
+  generateIdlNames, saveIdlNames,
+  generateIdlParsed, saveIdlParsed } = require('reffy');
+const {
+  createFolderIfNeeded,
+  requireFromWorkingDirectory,
+  copyFolder } = require('./utils');
+const { applyPatches } = require('./apply-patches.js');
+
+
+
+async function cleanCrawlOutcome(spec) {
+  await Promise.all(Object.keys(spec).map(async property => {
+    // Only consider properties that link to an extract
```

Comment:
since the list of extracted properties is more or less hardcoded in the rest of the code (e.g. L77), it might be better to use a shared hardcoded list rather than rely on heuristics?

Reply:
I went the opposite way, actually, and dropped the hardcoded list (except to note the folders that must not be integrated), so that the script can handle new extracts without having to be modified.

Note the heuristics are exactly the same as in the `expandCrawlResults` function in Reffy.